sklearn/utils/testing.py

```diff
@@ -422,6 +441,172 @@ def assert_raise_message(exceptions, message, function, *args, **kwargs):
                              (names, function.__name__))
 
 
+def _assert_same_model_method(method, X, estimator1, estimator2, msg=None):
+    method_err = '%r\n\nhas %s, but\n\n%r\n\ndoes not.'
+    if hasattr(estimator1, method):
+        if not hasattr(estimator2, method):
+            raise AssertionError(method_err % (estimator1, method, estimator2))
+
+        # Check if the method(X) returns the same for both models.
+        res1 = getattr(estimator1, method)(X)
+        res2 = getattr(estimator2, method)(X)
+
+        if sp.issparse(res1) or sp.issparse(res2):
+            same_model = _assert_sparse_allclose(res1, res2)
+        else:
+            same_model = ((res1.shape == res2.shape) and
+                          np.allclose(res1, res2))
```

Comment:
Yeah, I think you're right. It'd reduce the chance of introducing bugs, at least. And for small enough data, as we commonly use in tests, the slowdown might not even be significant.

Even if not, we could go with the densifying implementation first, and maybe later (when all common tests pass) speed it up if needed.